docs2/site/docs/getting-started/errors.md

```diff
@@ -1,78 +1,204 @@
-}
+    if (ctx.Exception is SqlException)
+        ctx.ErrorMessage = "A database error has occurred.";
+};
 ```
 
-You can provide additional error handling or logging for fields by adding Field Middleware.
+## Error Serialization
+
+After the `DocumentExecuter` has returned a `ExecutionResult` containing the data and/or errors,
+typically you will pass this object to an implementation of `IDocumentWriter` to convert the
+object tree into json. The GraphQL spec allows for four properties to be returned with each
+error: `message`, `locations`, `path`, and `extensions`. The `IDocumentWriter` implementations
+provided for the [`Newtonsoft.Json`](https://www.nuget.org/packages/GraphQL.NewtonsoftJson) and
+[`System.Text.Json`](https://www.nuget.org/packages/GraphQL.SystemTextJson) libraries allow you to control the
+serialization of `ExecutionError`s into the resulting json data by providing an `IErrorInfoProvider`
+to the constructor of the document writer. The `ErrorInfoProvider` class (default implementation of
+`IErrorInfoProvider`) contains 5 properties to control serialization behavior:
+
+* `ExposeExceptionStackTrace` when enabled sets the `message` property for errors to equal the
+exception's `.ToString()` method, which includes a stack trace. This property defaults to `false`.
+* `ExposeCode` when enabled sets the `extensions`'s `code` property to equal the error's `Code`
+property. This property defaults to `true`.
+* `ExposeCodes` when enabled sets the `extensions`'s `codes` property to equal a list containing both
+the error's `Code` property, if any, and the type name of inner exceptions (after being converted to
+UPPER_CASE and removing the "Extension" suffix). So an `ExecutionError` with a code of `INVALID_FORMAT`
+that has an inner exception of type `ArgumentNullException` would contain a `codes` property
+of `["INVALID_FORMAT", "ARGUMENT_NULL"]`. This property defaults to `true`.
+* `ExposeData` when enabled sets the `extension`'s `data` property to equal the data within the error's
+`Data` property. This property defaults to `true`.
+* `ExposeExtensions` when disabled hides the entire `extensions` property, including `code`, `codes`,
+and `data` (if enabled). This property defaults to `true`.
+
+For example, to show the stack trace for unhandled errors during development, you might write code like this:
+
+```csharp
+#if DEBUG
+    var documentWriter = new DocumentWriter(true, new ErrorInfoProvider(options => options.ExposeExceptionStackTrace = true));
+#else
+    var documentWriter = new DocumentWriter();
+#endif
+```
+
+You can also write your own implementation of `IErrorInfoProvider`. For instance, you might want to override
+the numerical codes provided by GraphQL.NET for validation errors, or reveal stack traces
+only to logged-in administrators.
+
+## Middleware
+
+It is also possible to provide additional error handling or logging for fields by adding Field
+Middleware, but this is not recommended.
```

Comment:
If you agree, I'd delete this section mentioning middleware.  I can't see how that would be beneficial to error handling over the other options available.

Reply:
OK